src/cli/commands/dev.ts

```diff
@@ -53,28 +57,32 @@ export class Dev implements Command {
       await p.hooks.dev.onStart?.()
     }
 
-    log.info('start', { version: ownPackage.version })
+    const runDebouncedReflection = async (layout: Layout.Layout) => {
+      const reflectionResult = await debouncedReflection(layout)
 
-    const runDebouncedReflection = simpleDebounce(async (layout: Layout.Layout) => {
-      const reflectionResult = await Reflection.reflect(layout, { artifacts: true })
+      if (reflectionResult.type === 'executing') {
+        return
```

Comment:
Doesn't this make it so the we see the error output from the not-last reflection run?

Reply:
No, `executing` will be returned if the function is called while another one is already running

Reply:
> will be returned if the function is called while another one is already running

Right, but the later call has the most up to date result.

Reply:
But I see problems with either approach here.

If we never print until there happens a completed reflection with no pending, then we may get in states where chains of reflection prevent user from seeing any feedback.

Reply:
Answered just below

Reply:
Ok I see now.